src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.Aot.cs

```diff
@@ -1037,4 +1043,86 @@ protected override void TransformNativeToManaged(ILCodeStream codeStream)
             StoreManagedValue(codeStream);
         }
     }
+
+    class OleCurrencyMarshaller : Marshaller
+    {
+        protected override void TransformManagedToNative(ILCodeStream codeStream)
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            LoadManagedValue(codeStream);
+
+            var helper = Context.GetHelperEntryPoint("InteropHelpers", "DecimalToOleCurrency");
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+
+            StoreNativeValue(codeStream);
+        }
+
+        protected override void TransformNativeToManaged(ILCodeStream codeStream)
+        {
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+            LoadNativeValue(codeStream);
+
+            var helper = Context.GetHelperEntryPoint("InteropHelpers", "OleCurrencyToDecimal");
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+
+            StoreManagedValue(codeStream);
+        }
+    }
+
+    class VariantMarshaller : Marshaller
+    {
+        protected override void AllocManagedToNative(ILCodeStream codeStream)
+        {
+            LoadNativeAddr(codeStream);
+            codeStream.Emit(ILOpcode.initobj, _ilCodeStreams.Emitter.NewToken(NativeType));
+        }
+
+        protected override void TransformManagedToNative(ILCodeStream codeStream)
+        {
+            if (this.MarshalDirection == MarshalDirection.Reverse)
+            {
+                throw new NotSupportedException();
+            }
+
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            LoadManagedValue(codeStream);
+            LoadNativeAddr(codeStream);
+
+            var helper = InteropTypes.GetMarshal(Context).GetKnownMethod("GetNativeVariantForObject", null);
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+        }
+
+        protected override void TransformNativeToManaged(ILCodeStream codeStream)
+        {
+            if (this.MarshalDirection == MarshalDirection.Reverse)
+            {
+                throw new NotSupportedException();
+            }
+
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            LoadNativeAddr(codeStream);
+
+            var helper = InteropTypes.GetMarshal(Context).GetKnownMethod("GetObjectForNativeVariant", null);
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(helper));
+
+            StoreManagedValue(codeStream);
+        }
+
+        protected override void EmitCleanupManaged(ILCodeStream codeStream)
+        {
+            // Only do cleanup if it is IN
```

Comment:
This comment should be other way around. This does cleanup only if it is not IN

Reply:
This is actually fine comment. We exit when `not IN`, so we `do cleanup` when IN. 

Interesting observation, that  negation is complicated for humans. I'm kind of interested in psychological properties of linguistic, so cannot resist take a note.

Reply:
Ah ok.